utilities/test_suite/HOST/Tensor_host_audio.cpp

```diff
@@ -197,6 +197,47 @@ int main(int argc, char **argv)
 
                     break;
                 }
+                case 6:
+                {
+                    testCaseName = "resample";
+                    Rpp32f inRateTensor[batchSize];
+                    Rpp32f outRateTensor[batchSize];
+
+                    maxDstWidth = 0;
+                    for(int i = 0; i < batchSize; i++)
+                    {
+                        inRateTensor[i] = 16000;
+                        outRateTensor[i] = 16000 * 1.15f;
+                        Rpp32f scaleRatio = outRateTensor[i] / inRateTensor[i];
+                        dstDims[i].width = (int)std::ceil(scaleRatio * srcLengthTensor[i]);
+                        dstDims[i].height = 1;
+                        maxDstWidth = std::max(maxDstWidth, (int)dstDims[i].width);
+                    }
+                    Rpp32f quality = 50.0f;
+                    Rpp32s lobes = std::round(0.007 * quality * quality - 0.09 * quality + 3);
+                    Rpp32s lookupSize = lobes * 64 + 1;
+                    RpptResamplingWindow window;
+                    windowed_sinc(window, lookupSize, lobes);
```

Comment:
Initially the windowed_sinc function was not passed a parameter
L217 to L220 in the test suite was inside the kernel function and we are generating the window class object inside the kernel

But this window class object is constant for all samples in the entire dataset.
Since RPP does batchwise processing the windowed_sinc function gets called for each batch in dataset and is recomputed. windowed_sinc function computation involved sinc and hann calculation 
https://github.com/snehaa8/rpp/blob/sn/resample/utilities/test_suite/rpp_test_suite_audio.h#L271

Rajy has asked previously to pass it as a parameter to avoid this recomputation for every batch, which is why this change has been made. Please let me know if there is any other workaround possible

Reply:
Okay, this computation would be part of rocAL then